### ISSUE TITLE
Composing Fields and using them throws error

### ISSUE THREAD

**Issue by joaovpmamede** [StoutLogic/acf-builder#8]
How you do register your [Reuse Example](https://github.com/StoutLogic/acf-builder#reuse-example)?
I'm trying to:

```
add_action('acf/init', function() use ($section) {
  acf_add_local_field_group($section->build());
});
```

but I'm getting a lot of warnings:
```
( ! ) Warning: Illegal string offset 'parent' in /srv/www/example.com/current/web/app/plugins/advanced-custom-fields-pro/pro/acf-pro.php on line 307
( ! ) Notice: Undefined index: type in /srv/www/example.com/current/web/app/plugins/advanced-custom-fields-pro/pro/acf-pro.php on line 288
```

**Comment by stevep** [comment]
@joaovpmamede that should work if you are wanting to register section field group. I will look into the notices and warnings.

**Comment by stevep** [comment]
So there were a few issues, one was a bug in the addFields method. That has been fixed. Hopefully your warnings and notices have gone away. If you are able to verify this with version `dev-master` let me know. I tested locally and was able to recreate your warnings and are now fixed on my end.

Secondly, there are some config errors I should correct in the README and Wiki

1. You do HAVE to set a location on the field groups. For some reason I assumed ACF defaulted to posts, but that is wrong. If you don't set a location, they won't show up anywhere. 
2.  In this example, the `columns` repeater field should be configured with a `['layout' => 'block']` or row. By default ACF will use the table layout, but this breaks the UI when tabs are used.

I'll make these documentation changes, and if you can verify things are fixed on your end with dev-master version, I can cut a new stable release.

Thanks so much for your help and interest!

**Comment by joaovpmamede** [comment]
Used the `dev-master` version and it seems to be working just fine.
I'll try a create a couple more examples and see if everything is okay.

No sir, I am the one who as to thank you for this awesome project.

**Comment by stevep** [comment]
Close in favor #12

**Comment by stevep** [comment]
Fixed in v1.0.4
